src/FrameSemantics.cc

```diff
@@ -282,289 +282,511 @@ static Errors resolveModelPoseWithPlacementFrame(
+
+  /// \brief Helper function to add children of interface models.
+  private: void AddInterfaceChildren(const sdf::InterfaceModel &_ifaceModel)
   {
-    auto nestedIfaceModel = _model->InterfaceModelByIndex(m);
-    if (outModel.Count(nestedIfaceModel->Name()) > 0)
+    for (const auto &item : _ifaceModel.Links())
     {
-      errors.push_back({ErrorCode::DUPLICATE_NAME,
-          "Nested interface model with non-unique name [" +
-              nestedIfaceModel->Name() + "] detected in model with name [" +
-              _model->Name() + "]."});
-      continue;
+      this->links.emplace_back(item);
+    }
+    for (const auto &item : _ifaceModel.Frames())
+    {
+      this->frames.emplace_back(item);
+    }
+    for (const auto &item : _ifaceModel.Joints())
+    {
+      this->joints.emplace_back(item);
+    }
+    for (const auto &item : _ifaceModel.NestedModels())
+    {
+      this->models.emplace_back(*item);
     }
-    auto nestedErrors = buildFrameAttachedToGraph(outModel, nestedIfaceModel);
-    errors.insert(errors.end(), nestedErrors.begin(), nestedErrors.end());
   }
+};
 
-  // add edges from joint to child frames
-  for (uint64_t j = 0; j < _model->JointCount(); ++j)
+/// \brief Wrapper for sdf::World
+struct WorldWrapper: public WrapperBase
+{
+  /// \brief Constructor that takes an sdf::World
+  explicit WorldWrapper(const sdf::World &_world)
+      : WrapperBase{_world.Name(), "World", FrameType::WORLD}
   {
-    auto joint = _model->JointByIndex(j);
-    auto jointId = outModel.VertexIdByName(joint->Name());
-    auto childFrameName = joint->ChildLinkName();
-    if (outModel.Count(childFrameName) != 1)
+    for (uint64_t i = 0; i < _world.FrameCount(); ++i)
     {
-      errors.push_back({ErrorCode::JOINT_CHILD_LINK_INVALID,
-        "Child frame with name[" + childFrameName +
-        "] specified by joint with name[" + joint->Name() +
-        "] not found in model with name[" + _model->Name() + "]."});
-      continue;
+      this->frames.emplace_back(*_world.FrameByIndex(i));
+    }
+    for (uint64_t i = 0; i < _world.ModelCount(); ++i)
+    {
+      this->models.emplace_back(*_world.ModelByIndex(i));
+    }
+    for (uint64_t i = 0; i < _world.InterfaceModelCount(); ++i)
+    {
+      this->models.emplace_back(*_world.InterfaceModelNestedIncludeByIndex(i),
+                                *_world.InterfaceModelByIndex(i));
     }
-    auto childFrameId = outModel.VertexIdByName(childFrameName);
-    outModel.AddEdge({jointId, childFrameId}, true);
   }
 
-  // add frame edges
-  for (uint64_t f = 0; f < _model->FrameCount(); ++f)
+  /// \brief Children frames.
+  std::vector<FrameWrapper> frames;
+  /// \brief Children models and interface models.
+  std::vector<ModelWrapper> models;
+};
+
+/////////////////////////////////////////////////
+/// \brief Add vertices to either Frame attached to or Pose graph. If the child
+/// element is a model, it calls the corresponding build*Graph function.
```

Comment:
Can we document the parameters? At a glance, I'm not sure what `items` are suppose to be

Reply:
Sure, added in https://github.com/ignitionrobotics/sdformat/pull/764/commits/59050f96134d6bf0cea07e35ab7916b1da7b70d9.